test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperDescriptorFactoryTest.cs

```diff
@@ -1755,6 +1779,14 @@ private class MultipleValidHtmlAttributePrefix : TagHelper
 
             [HtmlAttributeName("valid-name6")]
             public string StringProperty { get; set; }
+
+            public IDictionary<string, int> GetOnlyDictionaryProperty { get; }
+
+            [HtmlAttributeName("valid-name8")]
+            public IDictionary<string, object> GetOnlyDictionaryPropertyWithAttributeName { get; }
+
+            [HtmlAttributeName("ignored-name9", DictionaryAttributePrefix = "valid-prefix9")]
```

Comment:
Should also make the case of a `{ get; }` attribute with a non-null (or empty if you want, either way is fine with me) `HtmlAttributeName.Name` an error.